File: planik-android/app/src/main/java/pl/planik/local/mapper/to/model/PlanDayEntryEntityToModelMapper.kt
package pl.planik.local.mapper.to.model

import pl.planik.domain.model.DayEntry
import pl.planik.local.entity.PlanDayEntryEntity
import pl.planik.local.mapper.Mapper
import java.time.DayOfWeek
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PlanDayEntryEntityToModelMapper @Inject constructor() : Mapper<PlanDayEntryEntity, DayEntry> {
  override fun map(value: PlanDayEntryEntity): DayEntry {
    return DayEntry(
      id = value.id,
      dayOfWeek = DayOfWeek.of(value.dayOfWeek),
      title = value.title,
      start = value.start,
      end = value.end,
      pauseMinutes = 0
    )
  }
}

File: planik-flutter/planik/lib/local/mapper/plan_day_entry_entity_to_model_mapper.dart
import 'package:planik/domain/model/day_entry.dart';
import 'package:planik/local/entity/plan_day_entry_entity.dart';
import 'package:planik/local/mapper/mapper.dart';

class PlayDayEntryEntityToModelMapper implements Mapper<PlanDayEntryEntity, DayEntry> {
  const PlayDayEntryEntityToModelMapper();

  @override
  DayEntry map(PlanDayEntryEntity value) {
    return DayEntry(
      id: value.id!,
      dayOfWeek: value.dayOfWeek,
      title: value.name,
      start: DateTime.parse(value.start),
      end: DateTime.parse(value.end),
      pauseMinutes: 0,
    );
  }
}
